app.rb

```diff
@@ -0,0 +1,178 @@
+require './school_rental'
+require './school_person'
+require './school_student'
+require './school_teacher'
+require './school_book'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  def show_menu
+    puts "Welcome to School Library App!\n\n"
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts "7 - Exit\n\n"
+  end
+
+  def run
+    show_menu
+    option = gets.chomp
+
+    case option
+    when '1'
+      list_books
+    when '2'
+      list_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rental
+    else
+      puts 'Good-bye.'
+      exit
+    end
+  end
+
+  # 1
+  def list_books
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+      run
+    end
+  end
+
+  # 2
+  def list_people
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class.name}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      run
+    end
+  end
+
+  # 3
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    person_type = gets.chomp
+    case person_type
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'error'
+    end
+    puts "Person created successfully!\n"
+    run
+  end
+
+  # 3 (1)
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp
+
+    case permission
+    when 'y'
+      permission = true
+    when 'n'
+      permission = false
+    else
+      puts 'error'
+    end
+
+    student = Student.new(nil, age, name, permission)
```

Comment:
- [ ] Please I am having this error when I try to create student, checkout the screenshot below and kindly fix the issue.

![schoolLibraryError](https://user-images.githubusercontent.com/62849872/183998055-6a483762-74da-4f7d-8e47-163c3f51b725.png)